packages/framework/presence/src/presenceDatastoreManager.ts

```diff
@@ -256,9 +256,18 @@ export class PresenceDatastoreManagerImpl implements PresenceDatastoreManager {
 			// If it isn't then, not really a problem; just won't be in provider or quorum list.
 			// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 			const clientId = this.runtime.clientId!;
+			const requestor = message.clientId;
 			if (updateProviders.includes(clientId)) {
 				// Send all current state to the new client
 				this.broadcastAllKnownState();
+				this.presence.mc?.logger.sendTelemetryEvent({
+					eventName: "JoinResponse",
+					type: "broadcastAll",
+					// TODO: this is EUPI; how should it be tagged?
+					// If it can't be logged, then we can establish a correlation in the join request.
+					requestor,
+					role: "primary",
```

Comment:
I'd nest these under a property called `details`. It makes it easier to discover this info when scanning through a session log, and you can easily parse the JSON to operate/aggregate on the data if needed.

For our 1P partner's telemetry DB, we've run out of columns, so these will need to be pulled out of a catch-all anyway when querying - better to do it proactively / predictably.